doc/userguide/src/CreatingTestData/AdvancedFeatures.rst

```diff
@@ -739,16 +739,52 @@ to make the syntax easier to read.
        Repeat Keyword    42 times    My Keyword
        Repeat Keyword    ${var}    Another Keyword    argument
 
-Conditional execution
----------------------
+.. _if block:
+
+Conditional execution with IF
+-----------------------------
 
 In general, it is not recommended to have conditional logic in test
 cases, or even in user keywords, because it can make them hard to
 understand and maintain. Instead, this kind of logic should be in test
 libraries, where it can be implemented using natural programming
-language constructs. However, some conditional logic can be useful at
-times, and even though Robot Framework does not have an actual if/else
-construct, there are several ways to get the same effect.
+language constructs. However, conditional logic can be useful at
+times and from Robot Framework version 4.0 onwards IF block is included
```

Comment:
"IF block" doesn't sound that good to me but I'm not sure what would be better. Perhaps "IF syntax"?